pro/src/routes/User/adapters/patchIdentityAdapter.ts

```diff
@@ -0,0 +1,35 @@
+import { UserIdentityBodyModel, UserIdentityResponseModel } from 'apiClient/v1'
+
+import { api } from 'apiClient/api'
+
+type IPayloadFailure = null
+type PatchIdentityAdapter = Adapter<
+  UserIdentityBodyModel,
+  UserIdentityResponseModel,
+  IPayloadFailure
+>
+
+const FAILING_RESPONSE: AdapterFailure<IPayloadFailure> = {
+  isOk: false,
+  message:
+    'Une erreur est survenue lors de la récupération de vos informations',
+  payload: null,
+}
+
+const patchIdentityAdapter: PatchIdentityAdapter = async (
+  values: UserIdentityBodyModel
+) => {
+  try {
+    const payload = await api.patchUserIdentity(values)
+
+    return {
+      isOk: true,
+      message: '',
+      payload: payload,
+    }
+  } catch (error) {
+    return FAILING_RESPONSE
+  }
```

Comment:
Afin d'avoir acces aux erreurs de l'api dans la sortie de ton adapter il te faut les definir ici.
Au lieu d'utiliser un `FAILING_RESPONSE` générique tu aura: 

```suggestion
type IPayloadFailure = { [key: string]: string }
// ...
  } catch (error) {
    return {
      isOk: false,
      message:  'Une erreur est survenue lors de la récupération de vos informations',
      payload: { 
        errors
       }
    }
  }
```